cbv3_django_prototype/cbv3_django_prototype/resources/tests.py

```diff
@@ -72,25 +82,14 @@ def test_create_one_resource(self):
              "url":"https://javascript.info/",
              "referring_url":"https://gitconnected.com/learn/javascript",
              "other_referring_source": "iliakan@javascript.info",
-             "user":15,
              "date_published":"2019-09-19T03:27:06Z",
              "created":"2019-09-19T03:27:06.485Z",
              "modified":"2019-09-19T03:27:06Z",
              "media_type":"WEB",
-             "tags":[
```

Comment:
We **definitely** need tests for `tags` - and clearly, I've missed a whole nest of use cases.  :P .  I may file a completely separate issue for those rather than hold up this PR.

The reason `tags` don't appear in the `resources.json` file is a bit more complicated, and might prompt some larger discussions around how we manage tags in general across the system.  

While this endpoint interacts with tags **as if** they are part of the `Resources` model, they are in fact **not** stored in the `Resources` table, and aren't referred to by a FK from `Resources`.  

Instead, `taggit`s default configuration uses Djangos [**content types framework**](https://docs.djangoproject.com/en/3.0/ref/contrib/contenttypes/)  & a [**generic relation**](https://docs.djangoproject.com/en/3.0/ref/contrib/contenttypes/#generic-relations) to model a **manytomany** relationship.  

When the data is dumped directly from `Resources`, there is nothing in the `tags` field, because the `tags` field is managed by `taggable manager` code that takes care of the entries into the 3 associated tag tables (taggit_tag, taggit_taggeditem, django_content_type) and those associations are made by using the resources_resource t**content type**...but not a **key** from resources_resource.

As a consequence, there is now a `tags.json` fixture to load those associated tables into the DB because they can't be loaded directly from a `resources.json` without calling application code to do so.

Yes - that could very well be a setup for headaches later across endpoints (we are tagging literally everything), and I think it bears some concentrated discussion...like do we have a `Tags` endpoint?  Do we re-configure `taggit` to use a **through model** [**details here**](https://django-taggit.readthedocs.io/en/latest/custom_tagging.html)  to force explicit relations to each area?  what are the drawbacks of each approach?  What makes sense for how we'll be using this?  Because of the queries involved, will we be setting ourselves up for performance issues?

I'll stop now and open an issue.  :D

Reply:
Logged https://github.com/codebuddies/backend/issues/72 for `tags` tests.